crates/storage/src/db.rs

```diff
@@ -190,4 +196,37 @@ impl ReamDB {
             db: self.db.clone(),
         }
     }
+
+    pub fn is_initialized(&self) -> bool {
+        self.slot_index_provider()
+            .get_highest_slot()
+            .map_err(|err| anyhow!("Unable to fetch highest slot in db:{}", err))
+            .is_ok()
+    }
+}
+
+pub fn reset_db(path: Option<PathBuf>) -> anyhow::Result<()> {
+    let db_path = path.clone().ok_or_else(|| anyhow!("Path does not exist"))?;
+
+    if db_path.exists() {
+        info!(
+            "Are you sure you want to remove the database at {:?}? (y/n):",
+            path
+        );
+        io::stdout().flush()?;
+
+        let mut input = String::new();
+        io::stdin().read_line(&mut input).unwrap();
+        if input.trim().eq_ignore_ascii_case("y") {
+            match fs::remove_dir_all(&db_path) {
+                Ok(_) => info!("Database cleared successfully."),
+                Err(err) => return Err(anyhow!("Failed to remove database:{}", err)),
+            }
+        } else {
+            info!("Database path does not exist: {:?}", db_path);
+        }
+    } else {
+        info!("Operation canceled.");
+    }
+    Ok(())
```

Comment:
```suggestion
pub fn reset_db(data_dir: PathBuf) -> anyhow::Result<()> {
    if !fs::read_dir(&db_path)?.next().is_some(); {
        info!("Data directory at {db_path:?} is already empty.");
        return Ok(());
    }

    info!("Are you sure you want to clear the contents of the data directory at {db_path:?}? (y/n):");
    let mut input = String::new();
    io::stdin().read_line(&mut input)?;
    if input.trim().eq_ignore_ascii_case("y") {
        for entry in fs::read_dir(&db_path)? {
            let entry = entry?;
            let path = entry.path();
            if path.is_dir() {
                fs::remove_dir_all(&path)?;
            } else {
                fs::remove_file(&path)?;
            }
        }
        info!("Database contents cleared successfully.");
    } else {
        info!("Operation canceled by user.");
    }

    Ok(())
```

The data directory will always exist, the question is, is it empty or not?